### ISSUE TITLE
CASE with Named Params Ignored

### ISSUE THREAD

**Issue by jamisonhyatt** [kyleconroy/sqlc#1195]
### Version

1.10.0

### What happened?

I was playing around, writing code to work around https://github.com/kyleconroy/sqlc/issues/1155 with a `CASE` statement in `mysql`, and the generated code completely ignores the named parameters.  

edit: Ignore the validity of the schema/sql - I'm aware it janky - was just copy and pasting. :joy:

This could be the same failure to parse as https://github.com/kyleconroy/sqlc/issues/1194, but I wasn't sure and it felt better raised as 2 separate issues.

[Postgres working example](https://play.sqlc.dev/p/f4cb9747dd03fb212ac7c72d1bdd34d0e6d2442e59e486bbc89ae79d1d056cca)

### Database schema

```sql
CREATE TABLE authors (
  id   BIGINT PRIMARY KEY,
  username TEXT NULL,
  email TEXT NULL,
  name TEXT  NOT NULL,
  bio  TEXT,
  UNIQUE KEY idx_username (username),
  UNIQUE KEY ids_email (email)
);
```

### SQL queries

```sql
-- name: ListAuthors :one
SELECT * FROM authors
WHERE email = CASE WHEN sqlc.arg(email) = '' then NULL else sqlc.arg(email) END
OR username = CASE WHEN sqlc.arg(username) = '' then NULL else sqlc.arg(username) END 
LIMIT 1;
```

### Configuration

```yaml
{
  "version": "1",
  "packages": [
    {
      "path": "db",
      "engine": "mysql",
      "schema": "query.sql",
      "queries": "query.sql"
    }
  ]
}
```

### Playground URL

https://play.sqlc.dev/p/775c20fa89d28dc02643c0b8678d48a08bd58de32d24ae9ada73ffe9f157e5ab

### What database engines are you using?

MySQL

### What type of code are you generating?

Go

**Comment by kyleconroy** [comment]
Since this is probably the same issue as #1194, I'm going to be updating that issue instead of this one, but I'll keep this one open.

**Comment by kyleconroy** [comment]
Fixed in #1296
